src/main/java/patternverifiers/SingletonVerifier.java

```diff
@@ -1,17 +1,312 @@
+                    isStatic = false;
+                    isPrivate = false;
+                    return false;
+                }
+                if (!fieldDeclaration.getVariables().get(0).getInitializer().isEmpty()) {
+                    isInstantiated = true;
+                }
+                for (Modifier md : fieldDeclaration
+                    .getModifiers()) {     // For each modifier on that field
+                    if (md.getKeyword().asString().equals(
+                        "static")) {  // If that modifier is static
+                        isStatic = true;
+                    } else if (md.getKeyword().asString().equals(
+                        "private")) { // Else if that modifier is private
+                        isPrivate = true;
+                    }
+                }
+                instanceVar = fieldDeclaration;
+            }
+        }
+        return isStatic && isPrivate && isInstantiated; // should isInstansiated be here?
+    }
+
+    /**
+     * Method for declaring if a java class has a getInstance() method which calls the constructor
+     * of the Singleton class, does not support a Singleton that is initialized at variable
+     * declaration in the Instance field.
+     *
+     * @param compUnit The CompilationUnit representing the java class to look at
+     *
+     * @return
+     */
+    public boolean callsConstructor(CompilationUnit compUnit) {
+        boolean instanceMethod = true;
+        List<MethodDeclaration> methods = new ArrayList<>();
+        compUnit.findAll(MethodDeclaration.class).forEach(
+            methodDeclaration -> {  //Make a list of all
+                // methods in the class.
+                methods.add(methodDeclaration);
+            });
+        for (MethodDeclaration declaration : methods) { // For each method
+            if (declaration.getTypeAsString().equals(
+                compUnit.getPrimaryTypeName().get())) {   // If the method
+                // returns an instance of the Singleton
+                if (declaration.isStatic()) {  // If the method is static
+                    if (declaration.isPrivate()) {  // If the method is private
+                        compUnit.findAll(ConstructorDeclaration.class).forEach(
+                            constructor -> constDeclList.add(constructor));
+                        instanceMethod &= isMethodCalledFromPublic(methods, declaration);   //
+                        // Check if the method is called from a public method
+                        // (isMethodCalledFromPublic calls itself recursively to find nested
+                        // method calls)
+                        // return findConstructorCall(declaration, constructorDeclaration);
+                    } else {
+                        instanceMethod &= true; // AND with true
+                    }
+                } else {
+                    instanceMethod &= false;    // AND with false
```

Comment:
Shouldn't this be 
    instancemethod = false
for clarity?

Reply:
Resolved in next commit